WeatherApp/WeatherApp.Webpage/Controllers/ObservationController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WeatherApp.WebSite.Models;
+
+namespace WeatherApp.WebSite.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ObservationController : ControllerBase
+    {
+        readonly IObservationRepository _observationRepository;
+
+        public ObservationController(IObservationRepository observationRepository)
+        {
+            _observationRepository = observationRepository;
+        }
+
+        public object Observation { get; private set; }
+
+        [HttpGet("observations")]
+        public async Task<IEnumerable<Observation>> GetAllObservations()
+        {
+            var bob = await _observationRepository.Read();
+            return bob;
+        }
+
+        [HttpGet("{city}")]
+        public async Task<IEnumerable<Observation>> GetObservationsByCity(string city)
+        {
+            var observations = await _observationRepository.Read();
+            var observationsByCity =
+                from observation in observations
+                where observation.City.Equals(city)
+                select observation;
+
+            return observationsByCity.ToArray();
+        }
+
+        [HttpPost("{city}")]
+        [Consumes("application/x-www-form-urlencoded")]
+        public void Post([FromForm] IFormCollection formCollection, string city)
+        {
+            string userName = formCollection[nameof(userName)];
+            string description = formCollection[nameof(description)];
+            //string city = formCollection[nameof(city)];

```

Comment:
remove unnecessary code

Reply:
Thank you for your remark, it is fixed now.